### PR TITLE
Upgrade to hardhat 2.9 and fix mineBlocks scenario command

### PR DESCRIPTION
**Pull request by jflatow**

### CODE REVIEW

File: scenario/context/CometContext.ts [hunk 1]
```diff
@@ -244,11 +244,8 @@ export class CometContext {
     await this.world.hre.ethers.provider.send('evm_setNextBlockTimestamp', [timestamp]);
   }
 
-  // XXX Hardhat 2.9 has 'hardhat_mine' for mining multiple blocks
   async mineBlocks(blocks: number) {
-    for (let i = 0; i < blocks; i++) {
-      await this.world.hre.network.provider.send('evm_mine', []);
-    }
+    await this.world.hre.network.provider.send('hardhat_mine', [blocks]);
```
Comment (kevincheng96) [inline scenario/context/CometContext.ts:248]: I think `blocks` need to be of type hexstring. I'm seeing this error in you scenarios:

```
❌ upgrade Comet implementation and initialize [1 run]@mainnet: Error InvalidArgumentsError: Errors encountered in param 0: Invalid value 13140 supplied to : QUANTITY | undefined
    at validateParams (/home/runner/work/comet/comet/node_modules/hardhat/src/internal/core/jsonrpc/types/input/validation.ts:64:13)
    at HardhatModule._hardhatMineParams (/home/runner/work/comet/comet/node_modules/hardhat/src/internal/hardhat-network/provider/modules/hardhat.ts:385:26)
    at HardhatModule.processRequest (/home/runner/work/comet/comet/node_modules/hardhat/src/internal/hardhat-network/provider/modules/hardhat.ts:117:48)
    at HardhatNetworkProvider._send (/home/runner/work/comet/comet/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:211:35)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at runNextTicks (node:internal/process/task_queues:65:3)
    at listOnTimeout (node:internal/timers:528:9)
    at processTimers (node:internal/timers:502:7)
    at async HardhatNetworkProvider.request (/home/runner/work/comet/comet/node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:118:18)
```

**Review by kevincheng96 — COMMENTED** [review]
Awesome that you got this to work!

File: scenario/context/CometContext.ts (hunk 1 above)
Comment (kevincheng96) [reply scenario/context/CometContext.ts:248]: `blocks.toString(16)` should probably do the trick

Comment (jflatow) [reply scenario/context/CometContext.ts]: Fixed (also needs `0x` prefix)

**Review by kevincheng96 — APPROVED** [review]
💪

**Pull request merged**